safehttp/config.go

```diff
@@ -0,0 +1,24 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+// Config provide additional configurations to Interceptors when needed.
+type Config interface {
```

Comment:
Maybe `Option` would be a better name?

Reply:
We considered adding the options to the context. The drawback of this is that retrieving a value from the context is a linear search.

Reply:
This response was likely meant for another comment.

With regards to `Option` vs `Config`... I'd rather stick with `Config` for one specific reason here: it would seem that one `Interceptor` should have one `Config`, but could have multiple `Option`s. I'm leaning towards adjusting the interceptor's behavior just once.